### ISSUE TITLE
Are we over-engineering this?

### ISSUE THREAD

**Issue by mspraggs** [ShaneDrury/pyon#12]
This may be throwing a spanner in the works at this stage, but something has just occurred to me.

So Django has models, which describe the data we store, views, which take requests and models and create a response, urls, which pass requests to the views, and templates, which present the results.

We have models, which describe the data, and measurements, which take the data and do something with it before rendering it to some response. This seems awfully similar to the django architecture, so I wondered whether we should just use django, but hack it a bit to rename things.

Things that would be different:
- You still need Parser and a populatedb command that loads stuff into the database from your raw data files.
- Some hacking in the django server would be required so that urls were no longer the method for asking it to run measurements.
- Instead of returning an HTTP response, the views (now measurements) would do something like render the template and save it to disk, perhaps sending a success/failure status code back to the server.

Thoughts?

EDIT: Why not just have it so django creates a website for your project, then doing the measurements is just a case of following the various links. Or you could have forms that allow the user to select parameters for their measurements. You might need some clever AJAX to get a progress bar so that the page doesn't look like it's taking forever to load.

**Comment by ShaneDrury** [comment]
Yeah I like the idea about creating the links that perform the measurements. I think that's similar to my original plan of Pyon, where I wanted to make it easy to look at different aspects of the data say if your supervisor asked you to look at something you hadn't originally coded. I had envisaged the code somehow reasoning about your models/formulae and extracting as much as possible from it, but I don't think that would work past very simple cases. For example, my supervisor might ask me to bin the data and see how that changes results, but that's not trivial to code, so it's not realistic to have Pyon reason that from your code.
However, I would definitely still like to have the hyperlinks in the form of viewing extra data about the simulations e.g. plots but they wouldn't have to be AJAX since they could be generated along with the report and saved to disk.

So the question is if it's worth it to change the system to deal with requests and have this sort of website interface with a server running? I think it would be easier to just use Django outright and call views measurements rather than hack it to rename things. 
I can think of some advantages.
- It would be possible to implement a nice HTML interface to pass through measurement parameters/display results
- If you want to change e.g. the fit range you wouldn't have to edit the code, just change the values in the form
- Can make use of Django's cache framework so that repeat measurements would take less time to run

and some disadvantages:
- It would take a fair amount of work to code a basic measurement
- The end user would basically have to learn Django
- Pyon wouldn't actually do anything and would serve more as a library

It's a good point though, and I'm still thinking about it :)
